helix-core/src/main/java/org/apache/helix/controller/rebalancer/util/DelayedRebalanceUtil.java

```diff
@@ -311,30 +311,70 @@ public static Set<AssignableReplica> findToBeAssignedReplicasForMinActiveReplica
     Set<AssignableReplica> toBeAssignedReplicas = new HashSet<>();
 
     for (String resourceName : resources) {
-      // <partition, <state, instances set>>
-      Map<String, Map<String, Set<String>>> stateInstanceMap =
-          ClusterModelProvider.getStateInstanceMap(currentAssignment.get(resourceName));
       ResourceAssignment resourceAssignment = currentAssignment.get(resourceName);
-      String modelDef = clusterData.getIdealState(resourceName).getStateModelDefRef();
+      IdealState idealState = clusterData.getIdealState(resourceName);
+      String modelDef = idealState.getStateModelDefRef();
       Map<String, Integer> statePriorityMap = clusterData.getStateModelDef(modelDef).getStatePriorityMap();
+      ResourceConfig mergedResourceConfig =
+          ResourceConfig.mergeIdealStateWithResourceConfig(clusterData.getResourceConfig(resourceName), idealState);
```

Comment:
Is this because of user fixed assignment?

Reply:
It seems to be toward backward compatibility. I followed the convention to merge resource config, and it turns out resource config in clusterData might be null.